app/src/main/java/chat/rocket/android/chatroom/adapter/ChatRoomAdapter.kt

```diff
@@ -150,23 +150,23 @@ class ChatRoomAdapter(
     }
 
     fun prependData(dataSet: List<BaseUiModel<*>>) {
-        val item = dataSet.indexOfFirst { newItem ->
-            this.dataSet.indexOfFirst { it.messageId == newItem.messageId && it.viewType == newItem.viewType } > -1
-        }
-        if (item == -1) {
-            this.dataSet.addAll(0, dataSet)
-            notifyItemRangeInserted(0, dataSet.size)
-        } else {
-            dataSet.forEach { item ->
-                val index = this.dataSet.indexOfFirst {
-                    item.messageId == it.messageId && item.viewType == it.viewType
-                }
-                if (index > -1) {
-                    this.dataSet[index] = item
-                    notifyItemChanged(index)
-                }
+        //At first we will update all already saved elements with received updated ones
+        val filteredDataSet = dataSet.filter { newItem ->
+            val matchedIndex = this.dataSet.indexOfFirst { it.messageId == newItem.messageId && it.viewType == newItem.viewType }
+            if (matchedIndex > -1) {
+                this.dataSet[matchedIndex] = newItem
+                notifyItemChanged(matchedIndex)
             }
+            return@filter (matchedIndex < 0)
         }
+        //At the second stage we are inserting new received elements into set.
+        if (filteredDataSet.size == 0)
```

Comment:
Could you please put the `return` in the same line of `if (filteredDataSet.size == 0)`

Reply:
sure